### ISSUE TITLE
SWC: unknown field `noInterop`, expected `resolveFully`

### ISSUE THREAD

**Issue by meabed** [TypeStrong/ts-node#2070]
Using "@swc/core": "1.3.88", there is error with `noInterop`

```
JSON: {"sourceMaps":true,"module":{"noInterop":false,"type":"es6","strictMode":true,"ignoreDynamic":false},"swcrc":false,"jsc":{"parser":{"syntax":"typescript","tsx":false,"dynamicImport":true,"importAssertions":true},"target":"es2015","transform":{"legacyDecorator":true,"react":{"throwIfNamespace":false,"useBuiltins":false}},"keepClassNames":false,"experimental":{"keepImportAssertions":true}}}

Caused by:
    unknown field `noInterop`, expected `resolveFully` at line 1 column 391

```

**Comment by RomainLanz** [comment]
You have to stay to an older version of `@swc/core` or monkey patch the `ts-node` file until there is a new release that publishes the fix.

**Comment by JesseObrien** [comment]
Also having this issue with @swc/core @1.3.89.

**Comment by RomainLanz** [comment]
That's normal. The issue is not on `@swc/core` side but on `ts-node`. They have to push a release to make it work.

**Comment by niieani** [comment]
It would be great if there was a way to pass in SWC options from an ENV VAR or something to workaround those kinds of issues in the future.

**Comment by thetoxicavenger** [comment]
can we expect a fix on this soon? this blocks speeding up cdk apps via swc transpilation, unfortunately :/ https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk/README.md#changing-the-default-typescript-transpiler

**Comment by thetoxicavenger** [comment]
> You have to stay to an older version of `@swc/core` or monkey patch the `ts-node` file until there is a new release that publishes the fix.

@RomainLanz do you know what version works?

**Comment by thetoxicavenger** [comment]
> ###
> **[thetoxicavenger](/thetoxicavenger) ** commented [Sep 27, 2023](#issuecomment-1737972543)

just checked, `1.3.82` works

**Comment by RomainLanz** [comment]
Yes, I am also forcing to `1.3.82`.

You can also use [`patch-package`](https://www.npmjs.com/package/patch-package) to monkey patch the dependency on the fly.

**Comment by jtmueller** [comment]
> That's normal. The issue is not on `@swc/core` side but on `ts-node`. They have to push a release to make it work.

I agree that an update to ts-node is the most practical fix, but it kind of looks like swc broke backwards compatibility by throwing an error when it encounters a no-longer-supported option, when it should have silently ignored that option.

Why is `"noInterop":false` being treated any differently than not specifying `noInterop`? Logically the outcome is the same, so it looks like an swc bug - it's not ts-node's fault that swc broke backwards-compatibilty.

**Comment by javadoug** [comment]
The SWC should have released server major to prevent breaking all ts-node consumers who use swc.

Workaround for me was to use resolutions and overrides in package.json to force downgrade of all transient dependencies of @swc/core until a ts-node can release with the update.

```
{
    resolutions: { "@swc/core": "1.2.82" },
    overrides: { "@swc/core": "1.2.82" }
}
```

**Comment by jbinard** [comment]
It looks like https://github.com/TypeStrong/ts-node/pull/2062 fixes this issue but there was no new release since July. Can you publish a new release to fix it ?

**Comment by dgadelha** [comment]
If anybody doesn't need any specific ts-node features, install the `@swc-node/register` package and replace ts-node with it according to the [documentation](https://github.com/swc-project/swc-node#usage):

```
npm i -D @swc-node/register
node -r @swc-node/register script.ts
node --loader @swc-node/register/esm script.ts # for esm project
```

**Comment by kkimdev** [comment]
@cspotcode Friendly ping?

**Comment by jacoblee93** [comment]
Bump as well!

**Comment by fohanist** [comment]
It does not work with the following versions either.

```json
{
  "@swc/core": "^1.3.93",
  "ts-node": "^10.9.1",
}
```

**Comment by xobotyi** [comment]
same here

**Comment by przemyslawzalewski** [comment]
FYI: The latest @swc/core (at least [^1.3.94](https://github.com/swc-project/swc/blob/main/CHANGELOG.md#1394---2023-10-21)) no longer errors out on the unknown fields as [the above issue](https://github.com/swc-project/swc/issues/7979) has been resolved via https://github.com/swc-project/swc/pull/8163 🎉
